flare/gp.py

```diff
@@ -213,14 +219,21 @@ def check_instantiation(self):
 
         self.bounds = deepcopy(self.hyps_mask.get('bounds', None))
 
-    def update_kernel(self, kernels, component="mc", hyps_mask=None):
+    def update_kernel(self, kernels: List[str], component: str = "mc", hyps_mask: dict=None):
         kernel, grad, ek, efk, _, _, _ = str_to_kernel_set(
             kernels, component, hyps_mask)
         self.kernel = kernel
         self.kernel_grad = grad
         self.energy_force_kernel = efk
         self.energy_kernel = ek
         self.kernels = kernel_str_to_array(kernel.__name__)
+        
+        if hyps_mask is not None:
+            self.hyps_mask = hyps_mask
+            if self.cutoffs!= hyps_mask['cutoffs']:
```

Comment:
maybe make cutoff as an argument instead? hyps_mask does not necessarily have a cutoff argument.